[Develop] Use names for tutorial URL ids

rschamp opened the pull request:
To make URLs more friendly, use e.g., `tutorial=getting-started` instead of `tutorial=1`.  This will also help with some URL handling we're doing for the new project page.

kchadha commented on src/lib/tutorial-from-url.js:
```diff
@@ -36,15 +36,10 @@ const getDeckIdFromUrlId = urlId => {
+    const tutorialID = Array.isArray(queryParams.tutorial) ?
+        queryParams.tutorial[0] :
+        queryParams.tutorial;
+    return getDeckIdFromUrlId(tutorialID);
```
Should we also add an early exit in getDeckIdFromUrlId if tutorialID is `undefined`...?

kchadha replied:
Would a tutorial that is missing the 'urlId' field get matched to this tutorialId in this case?

rschamp replied:
Good catch, we should exit early!

chrisgarrity commented on src/lib/libraries/decks/index.jsx:
```diff
@@ -423,7 +423,7 @@ export default {
         }
         ],
-        urlId: 2
+        urlId: 'animate-a-name'
```
Currently: `projects/editor/?tip_bar=name`

chrisgarrity commented on src/lib/libraries/decks/index.jsx:
```diff
@@ -492,7 +492,7 @@ export default {
         }
         ],
-        urlId: 3
+        urlId: 'make-music'
```
`/projects/editor/?tip_bar=music`

rschamp commented:
@chrisgarrity I found one more, `hide`. But none of the rest seem to map, so I left them alone. Happy to take any other suggestions.

@kchadha I fixed that bug and added a regression test.

thisandagain commented:
This looks good. Thanks @rschamp !

chrisgarrity commented:
👍
